test/testdata/parser/error_recovery/block_arg_and_block.rb

```diff
@@ -0,0 +1,6 @@
+# typed: true
+
+[1,2,3].each do |x|
+  foo(&bar) do end
```

Comment:
Idk whether you want to include any code in the do/end to show that we don’t discard it either.